### PR TITLE
scripts: hid_configurator: Support pure ED25519 signature

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds support for pure ED25519 signature (used by nRF54L based devices that enable MCUboot hardware cryptography). imgtool from MCUboot upstream repository does not support this configuration, a dedicated imgtool version from sdk-mcuboot repository must be used.

Jira: NCSDK-30745

### CODE REVIEW

File: scripts/hid_configurator/README.rst [hunk 1]
```diff
@@ -91,6 +91,18 @@ Complete the following steps:
...
+   The ``imgtool`` from PyPI currently does not support pure ED25519 signature (used by nRF54L based devices that enable MCUboot hardware cryptography).
+   This may result in rejecting proper DFU images (``DFU image is invalid``).
+   ``imgtool`` supporting pure ED25519 signature can be installed from ``sdk-mcuboot`` repository (:file:`ncs/bootloader/mcuboot/scripts` directory of the |NCS|).
+   Run the following commands in the source directory to install ``imgtool`` supporting pure ED25519 signature and the required dependencies:
+
+   .. parsed-literal::
+      :class: highlight
+
+      py -3 -m pip install -r requirements.txt
+      py -3 -m pip install .
```
Comment (nordicjm) [inline scripts/hid_configurator/README.rst:104]: not that I know about python but why can't it do as the build system does and use the in-tree version until a proper updated version is released? See https://github.com/nrfconnect/sdk-nrf/blob/main/cmake/sysbuild/image_signing.cmake#L52

Comment (MarekPieta) [reply scripts/hid_configurator/README.rst:104]: This step actually installs the in-tree version of `imgtool` instead of version from PyPI. Python script by itself has no knowledge where `sdk-mcuboot` repository sources are on host computer - that's why I decided to install proper `imgtool` through pip. Script could be moved, so using a relative path might be problematic too.

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -646,7 +646,14 @@ Scripts
 
 This section provides detailed lists of changes by :ref:`script <scripts>`.
 
-* Added semantic version support to :ref:`nrf_desktop_config_channel_script` Python script for devices that use the SUIT DFU.
```
Comment (kapi-no) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:649]: What happened here? @divipillai 

This entry was already mentioned in NCS 2.8 as I added it:

https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/releases_and_maturity/releases/release-notes-2.8.0.html#scripts

File: scripts/hid_configurator/README.rst (hunk 1 above)
Comment (kapi-no) [reply scripts/hid_configurator/README.rst:104]: The user should already have the correct version if they follow the NCS installation guide correctly. We could minimize this note and notify the user to double-check if they have the correct version of imgtool.

We can also add a reference to the NCS install documentation that covers how to install this tool:

https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/installation/install_ncs.html#system-wide_installation
(Step 4)

File: scripts/hid_configurator/modules/dfu.py [hunk 3]
```diff
@@ -209,7 +209,13 @@ def b0_get_dfu_image_bootloader_var():
 
 
 def mcuboot_common_is_dfu_file_correct(dfu_bin):
-    res, _, _ = imgtool.image.Image.verify(dfu_bin, None)
+    try:
+        res, _, _ = imgtool.image.Image.verify(dfu_bin, None)
```
Comment (kapi-no) [inline scripts/hid_configurator/modules/dfu.py:213]: should we still support the outdated `verify` API?

File: scripts/hid_configurator/README.rst (hunk 1 above)
Comment (MarekPieta) [reply scripts/hid_configurator/README.rst:104]: The nRF54L MCuboot's pure ED25519 signature requires custom imgtool version (not from PyPI; needs to be installed from source using `sdk-mcuboot` repo). Unless I missed something, the custom imgtool version is not installed together with our official toolchain. I am not sure if there is better way of using the custom imgtool in this context than manually installing from source (as currently suggested). Please propose alternative solutions if anything better comes to your mind.

I hope that this is only a temporary workaround and could be removed when https://github.com/mcu-tools/mcuboot/pull/2063 is introduced to imgtool from PyPI. @de-nordic and @nvlsianpu, do you think there is a better way of doing it for now?

File: scripts/hid_configurator/modules/dfu.py (hunk 3 above)
Comment (MarekPieta) [reply scripts/hid_configurator/modules/dfu.py:213]: It's not "outdated" but official". As a current workaround, we catch exception and retry with 4 returned values, because  imgtool version from `sdk-mcuboot` broke backwards compatibility and returns 4 values instead of 3

File: scripts/hid_configurator/README.rst (hunk 1 above)
Comment (MarekPieta) [reply scripts/hid_configurator/README.rst:104]: @fundakol, could you advise? (@de-nordic recommended to ask you)

Comment (nordicjm) [reply scripts/hid_configurator/README.rst:104]: Comment from Carles: "you just add the line where your extra script is into the system path"

Comment (nordicjm) [reply scripts/hid_configurator/README.rst:104]: See how it is done e.g.
```
doc/internal/conf.py:sys.path.insert(0, str(NRF_BASE / "doc" / "_utils"))
doc/internal/conf.py:sys.path.insert(0, str(ZEPHYR_BASE / "doc" / "_extensions"))
doc/internal/conf.py:sys.path.insert(0, str(NRF_BASE / "doc" / "_extensions"))
doc/kconfig/conf.py:sys.path.insert(0, str(NRF_BASE / "doc" / "_utils"))
doc/kconfig/conf.py:sys.path.insert(0, str(ZEPHYR_BASE / "doc" / "_extensions"))
```

File: scripts/hid_configurator/modules/dfu.py (hunk 3 above)
Comment (kapi-no) [reply scripts/hid_configurator/modules/dfu.py:213]: The official is the one that we refer to in the installation guide. The PyPI version is irrelevant, as you always get the correct version if you set up the NCS correctly or according the installation instructions

Comment (MarekPieta) [reply scripts/hid_configurator/modules/dfu.py:213]: "official" is the one from PyPI (https://pypi.org/project/imgtool/)
Temporarily, we need to use imgtool from source (https://github.com/nrfconnect/sdk-mcuboot/tree/main/scripts) to support pure ED25519 (needed for our nRF54L use-case). Unless I missed something, setting up NCS would not result in installing the "from source" version automatically.

Comment (kapi-no) [reply scripts/hid_configurator/modules/dfu.py:213]: discussed offline, irrelevant

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (peknis) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:649]: That was in 2.8.0, so I guess we do not need that entry for 2.9.0. Please add the new stuff as a comment to my https://github.com/nrfconnect/sdk-nrf/pull/19240 and remove from this PR.

Comment (MarekPieta) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:649]: I will fix this (in separate PR). I will consult with @divipillai

File: scripts/hid_configurator/README.rst (hunk 1 above)
Comment (MarekPieta) [reply scripts/hid_configurator/README.rst:104]: Then we need to know `NRF_BASE`/`ZEPHYR_BASE` in scope of the script (it might become problematic e.g. if someone would like to use script outside of the nRF Connect SDK as a standalone tool). Also, the sdk-mcuboot's `imgtool`'s scripts would need to be located in a known place.

Please keep in mind that we use use `image` class exposed by the imgtool package (we do not call imgtool through CLI). Because of that, IMHO relying on packages might be cleaner approach - we build a SUIT-related package from source and install it manually already too.

Also please keep in mind that targets other than nRF54L do not use pure ED25519 signature, so they may still rely on the default imgtool version (extra step from the note could be basically skipped). Maybe we could explicitly comment on that and stick with using package built from source as a temporary workaround (we will use imgtool from PyPI as soon as it starts supporting the pure signature)?

Comment (pdunaj) [reply scripts/hid_configurator/README.rst:104]: @MarekPieta , I don't see @nordicjm blocking the change. I value input from various reviewers but this area is not his responsibility. Please keep that in mind.

I personally think that provided toolchain should cover all things needed. If it does not we should request it from team responsible for toolchain - check if we or pluto have a ticket. In the end this tool is not only our problem.

Since the timing is short I expect having the workaround is valid. It will require user interaction and is part of nrf_desktop tools doc only.

**Review by pdunaj — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (divipillai) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:649]: My bad, I missed to clean this entry while cleaning the changelog. Checked the Change log cleanup PR again and this is the only entry I missed :) . Good catch. Since we are already removing the entry in this PR, I think no action is required here.

Comment (peknis) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:649]: Some mishap during the releasing I guess. It was the first time we had the preview doc release and releasing from branch.

File: scripts/hid_configurator/README.rst (hunk 1 above)
Comment (fundakol) [reply scripts/hid_configurator/README.rst:104]: @MarekPieta, @nordicjm, command `pip install .` just installs Python's package from a source code, if that is what you want it is fine for me. I would only add a link to the repository where is imgtool (I don't see it in the readme file, and user may not know how to find it). 

Also you can install directly from a git repository like so:
`pip install git+ssh://git@github.com/myuser/foo.git@branch_or_tag#subdirectory=stackoverflow`
Which could be more convenient for users.

Comment (MarekPieta) [reply scripts/hid_configurator/README.rst:104]: I provided source directory path starting from nRF Connect SDK root directory (should be straightforward to find it for an NCS user). I do not provide a complete path in command here as I cannot provide absolute path and relative path might change in case user moves HID configurator scripts locally. Would this approach be ok?

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (kapi-no) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:649]: Thanks for the explanation. We are not removing this entry in this PR but @MarekPieta promised to do that in a follow-up PR. @MarekPieta, if you plan to update this PR, you can consider adding a separate commit that remove this outdated entry

**Review by kapi-no — APPROVED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (MarekPieta) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Eventually will be removed as part of @peknis PR: https://github.com/nrfconnect/sdk-nrf/pull/19240#discussion_r1869608845?

Comment (kapi-no) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: @MarekPieta, the latest changelog is not changed in @peknis PR

Comment (kapi-no) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: oh I think the changelog reset will happen as part of the following PR:

https://github.com/nrfconnect/sdk-nrf/pull/19241

Comment (peknis) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Due to the releasing-from-branch approach, there will be two resets. In both 19241 and 19240. Complicated and new to me as well. You better discuss with Divya as she has a bit of experience already.

File: scripts/hid_configurator/README.rst [hunk 4]
```diff
@@ -151,6 +163,17 @@ Complete the following steps:
       pip3 install --user -r requirements.txt
       pip3 install --user .
 
+.. note::
+   The ``imgtool`` from PyPI currently does not support pure ED25519 signature (used by nRF54L based devices that enable MCUboot hardware cryptography).
```
Comment (peknis) [inline scripts/hid_configurator/README.rst:167]:
```suggestion
   Currently, the ``imgtool`` from PyPI does not support pure ED25519 signature (used by nRF54L-based devices that enable MCUboot hardware cryptography).
```

File: scripts/hid_configurator/README.rst [hunk 5]
```diff
@@ -151,6 +163,17 @@ Complete the following steps:
...
+   This may result in rejecting proper DFU images (``DFU image is invalid``).
+   ``imgtool`` supporting pure ED25519 signature can be installed from ``sdk-mcuboot`` repository (:file:`ncs/bootloader/mcuboot/scripts` directory of the |NCS|).
+   Run the following commands in the source directory to install ``imgtool`` supporting pure ED25519 signature and the required dependencies:
```
Comment (peknis) [inline scripts/hid_configurator/README.rst:170]:
```suggestion
   Run the following commands in the source directory to install ``imgtool`` and the required dependencies:
```

File: scripts/hid_configurator/README.rst [hunk 6]
```diff
@@ -91,6 +91,18 @@ Complete the following steps:
       py -3 -m pip install -r requirements.txt
       py -3 -m pip install .
 
+.. note::
+   The ``imgtool`` from PyPI currently does not support pure ED25519 signature (used by nRF54L based devices that enable MCUboot hardware cryptography).
```
Comment (peknis) [inline scripts/hid_configurator/README.rst:95]:
```suggestion
   Currently, the ``imgtool`` from PyPI does not support pure ED25519 signature (used by nRF54L-based devices that enable MCUboot hardware cryptography).
```

File: scripts/hid_configurator/README.rst [hunk 7]
```diff
@@ -91,6 +91,18 @@ Complete the following steps:
...
+   This may result in rejecting proper DFU images (``DFU image is invalid``).
+   ``imgtool`` supporting pure ED25519 signature can be installed from ``sdk-mcuboot`` repository (:file:`ncs/bootloader/mcuboot/scripts` directory of the |NCS|).
+   Run the following commands in the source directory to install ``imgtool`` supporting pure ED25519 signature and the required dependencies:
```
Comment (peknis) [inline scripts/hid_configurator/README.rst:98]:
```suggestion
   Run the following commands in the source directory to install ``imgtool`` and the required dependencies:
```

**Review by peknis — DISMISSED** [review]

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst (hunk 2 above)
Comment (divipillai) [reply doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst]: Yeah, your understanding here is correct @kapi-no .

The Changelog will be rest in https://github.com/nrfconnect/sdk-nrf/pull/19241 and the new entry for the 2.9 release could be added to https://github.com/nrfconnect/sdk-nrf/pull/19240 , which is already done. I think its sorted now.

File: scripts/hid_configurator/README.rst [hunk 8]
```diff
@@ -91,6 +91,18 @@ Complete the following steps:
       py -3 -m pip install -r requirements.txt
       py -3 -m pip install .
 
+.. note::
+   Currently, the ``imgtool`` from PyPI does not support pure ED25519 signature (used by nRF54L-based devices that enable MCUboot hardware cryptography).
+   This may result in rejecting proper DFU images (``DFU image is invalid``).
+   ``imgtool`` supporting pure ED25519 signature can be installed from ``sdk-mcuboot`` repository (:file:`ncs/bootloader/mcuboot/scripts` directory of the |NCS|).
```
Comment (divipillai) [inline scripts/hid_configurator/README.rst:97]:
```suggestion
   ``imgtool`` supporting pure ED25519 signature can be installed from the ``sdk-mcuboot`` repository (:file:`ncs/bootloader/mcuboot/scripts` directory of the |NCS|).
```

File: scripts/hid_configurator/README.rst [hunk 9]
```diff
@@ -151,6 +163,17 @@ Complete the following steps:
       pip3 install --user -r requirements.txt
       pip3 install --user .
 
+.. note::
+   Currently, the ``imgtool`` from PyPI does not support pure ED25519 signature (used by nRF54L-based devices that enable MCUboot hardware cryptography).
+   This may result in rejecting proper DFU images (``DFU image is invalid``).
+   ``imgtool`` supporting pure ED25519 signature can be installed from ``sdk-mcuboot`` repository (:file:`ncs/bootloader/mcuboot/scripts` directory of the |NCS|).
```
Comment (divipillai) [inline scripts/hid_configurator/README.rst:169]:
```suggestion
   ``imgtool`` supporting pure ED25519 signature can be installed from the ``sdk-mcuboot`` repository (:file:`ncs/bootloader/mcuboot/scripts` directory of the |NCS|).
```

File: scripts/hid_configurator/README.rst [hunk 10]
```diff
@@ -151,6 +163,17 @@ Complete the following steps:
...
+   Run the following commands in the source directory to install ``imgtool`` and the required dependencies:
```
Comment (divipillai) [inline scripts/hid_configurator/README.rst:170]: We could add these texts as includes as these are repetition of line nos : 94-98. (This can be taken care in a separate PR as well).

.. include:: 
   :start-after
   :end-before:

**Review by divipillai — APPROVED** [review]
